front/components/assistant/VoicePicker.tsx

```diff
@@ -0,0 +1,319 @@
+        description: "Your voice message was added to attachments.",
+      });
+    },
+    [fileUploaderService, sendNotification]
+  );
+
+  const clearPressTimeout = () => {
+    if (pressTimeoutRef.current !== null) {
+      window.clearTimeout(pressTimeoutRef.current);
+      pressTimeoutRef.current = null;
+    }
+  };
+
+  // ----- Recording flow -------------------------------------------------------
+
+  const startRecording = useCallback(async () => {
+    if (isRecording) {
+      return;
+    }
+    try {
+      const stream = await requestMicrophone();
+      streamRef.current = stream;
+
+      const recorder = createRecorder(stream, chunksRef);
+      mediaRecorderRef.current = recorder;
+
+      recorder.start();
+      setIsRecording(true);
+    } catch {
+      sendNotification({
+        type: "error",
+        title: "Microphone permission required.",
+        description: "Please allow microphone access and try again.",
+      });
+    }
+  }, [isRecording, sendNotification]);
+
+  const stopAndFinalize = useCallback(
+    async (reason: "hold" | "click") => {
+      setIsRecording(false);
+      setIsTranscribing(true);
+
+      try {
+        const file = buildAudioFile(chunksRef.current);
+        chunksRef.current = [];
+
+        if (reason === "hold") {
+          await finalizeRecordingHold(file);
+        } else {
+          await finalizeRecordingClick(file);
+        }
+      } catch (e) {
+        sendNotification({
+          type: "error",
+          title: "Recording error.",
+          description:
+            e instanceof Error ? e.message : "An unknown error occurred.",
+        });
+      } finally {
+        setIsTranscribing(false);
+        stopTracks(streamRef.current);
+        streamRef.current = null;
+      }
+    },
+    [finalizeRecordingHold, finalizeRecordingClick, sendNotification]
+  );
+
+  const stopRecording = useCallback(
+    async (reason: "hold" | "click") => {
+      const recorder = mediaRecorderRef.current;
+      if (!recorder || recorder.state === "inactive") {
+        return;
+      }
+
+      const stopped = new Promise<void>((resolve) => {
+        recorder.onstop = () => resolve();
+      });
+
+      recorder.stop();
+      await stopped;
+      await stopAndFinalize(reason);
+    },
+    [stopAndFinalize]
+  );
+
+  // ----- Event handlers -------------------------------------------------------
+
+  const handlePointerDown = async () => {
+    if (disabled) {
+      return;
+    }
+    // Mark the start time of the press.
+    pressStartRef.current = Date.now();
+
+    // Schedule transition to hold mode if press lasts >= 150ms.
+    clearPressTimeout();
+    pressTimeoutRef.current = window.setTimeout(async () => {
+      // If still pressed, enter hold mode and start recording.
+      if (pressStartRef.current !== null) {
+        setMode("hold");
+        if (!isRecording) {
+          await startRecording();
+        }
+      }
+    }, 150);
+  };
```

Comment:
this is the main handling of hold vs click

the browser will send onClick and handlePointerDown/Up each time a button is clicked, whatever the time the mouse was pressed down. 

so we need an heuristic to know if the user wanted to "hold" or "click". 150ms seemed a good threshold